agent/consul/server.go

```diff
@@ -1315,6 +1315,7 @@ func (s *Server) setupExternalGRPC(config *Config, typeRegistry resource.Registr
 		Logger:      logger.Named("grpc-api.dataplane"),
 		ACLResolver: s.ACLResolver,
 		Datacenter:  s.config.Datacenter,
+		EnableV2:    stringslice.Contains(deps.Experiments, catalogResourceExperimentName),
```

Comment:
Did we miss injecting the ResourceAPIClient here, or will that come later?

Reply:
Yep I missed it. Great catch!